pallets/capacity/src/lib.rs

```diff
@@ -1062,6 +1068,80 @@ impl<T: Config> Pallet<T> {
 		ProviderBoostHistories::<T>::set(account, Some(boost_history));
 		Ok(())
 	}
+
+	pub(crate) fn has_unclaimed_rewards(account: &T::AccountId) -> bool {
+		let current_era = Self::get_current_era().era_index;
+		match Self::get_staking_history_for(account) {
+			Some(provider_boost_history) => {
+				match provider_boost_history.count() {
+					0usize => false,
+					// they staked before the current era, so they have unclaimed rewards.
+					1usize => provider_boost_history.get_entry_for_era(&current_era).is_none(),
+					_ => true,
+				}
+			},
+			None => false,
+		} // 1r
+	}
+
+	// this could be up to 35 reads.
```

Comment:
What about making the storage of past eras a map of bounded vecs of for example size `10`?

Thus this key-value data structure: `<floor(block number /10)>: <10 blocks worth of data>`

It limits the size of the write when the epoch changes, but for 99.9% of users would limit the data to pull when claiming.

Reply:
I don't quite understand what you mean. And also, are you talking about individual boost history, or overall RewardPool history?

Reply:
I was thinking the the overall history: `StakingRewardPool`. If the issue is that having each era separate creates too many reads, and having all of them in one is too much to read every time, then perhaps we could find a chunking setup that balances between the two. Chunking on the `era % 10` for example.

Reply:
Ok, as discussed previously, I'll try this idea out in a separate branch after this one is merged.